orcid_hub/models.py

```diff
@@ -1528,47 +1590,53 @@ def val(row, i, default=None):
+        for r in records:
+            validation_source_data = copy.deepcopy(r)
+            validation_source_data = del_none(validation_source_data)
 
-                # Adding schema valdation for funding
-                validator = Core(
-                    source_data=validation_source_data,
-                    schema_files=[os.path.join(SCHEMA_DIR, "funding_schema.yaml")])
-                validator.validate(raise_exception=True)
+            # Adding schema valdation for funding
+            validator = Core(
+                source_data=validation_source_data,
+                schema_files=[os.path.join(SCHEMA_DIR, "funding_schema.yaml")])
+            validator.validate(raise_exception=True)
 
+        with db.atomic():
             try:
                 if org is None:
                     org = current_user.organisation if current_user else None
-                task = Task.create(org=org, filename=filename, task_type=TaskType.FUNDING)
-
-                for funding_data in funding_data_list:
-
-                    title = get_val(funding_data, "title", "title", "value")
-                    translated_title = get_val(funding_data, "title", "translated-title", "value")
-                    translated_title_language_code = get_val(funding_data, "title", "translated-title", "language-code")
-                    type = funding_data.get("type")
-                    organization_defined_type = get_val(funding_data, "organization-defined-type", "value")
-                    short_description = funding_data.get("short-description")
-                    amount = get_val(funding_data, "amount", "value")
-                    currency = get_val(funding_data, "amount", "currency-code")
-                    start_date = PartialDate.create(funding_data.get("start-date"))
-                    end_date = PartialDate.create(funding_data.get("end-date"))
-                    org_name = get_val(funding_data, "organization", "name")
-                    city = get_val(funding_data, "organization", "address", "city")
-                    region = get_val(funding_data, "organization", "address", "region")
-                    country = get_val(funding_data, "organization", "address", "country")
-                    disambiguated_org_identifier = get_val(funding_data, "organization", "disambiguated-organization",
-                                                           "disambiguated-organization-identifier")
-                    disambiguation_source = get_val(funding_data, "organization", "disambiguated-organization",
-                                                    "disambiguation-source")
+                if not task:
+                    task = Task.create(org=org, filename=filename, task_type=TaskType.FUNDING)
+                else:
+                    FundingRecord.delete().where(FundingRecord.task == task).execute()
+
+                for r in records:
+
+                    title = r.get("title", "title", "value")
```

Comment:
i guess, the yaml funding upload will break, as the output of yaml.load(source) is not NestedDict